app/javascript/packs/AddNewRestPopup.jsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import $ from 'jquery';
+
+class AddNewRestPopup extends React.Component {
+
+    constructor() {
+        super();
+        this.state = {
+            name: "",
+            address: "",
+            tenbis: false,
+            max_delivery_time: 10,
+            cuisine: ""
+        };
+
+        this.updateChange = this.updateChange.bind(this);
+        this.newRestaurantSubmit = this.newRestaurantSubmit.bind(this);
+    }
+
+    updateChange(e) {
+        const event_id = e.target.id;
+        const event_value = e.target.value;
+
+        if (event_id === "restaurant_name") {
+            this.setState({name: event_value});
+        }
+        else if (event_id === "restaurant_address") {
+            this.setState({address: event_value});
+        }
+        else if (event_id === "restaurant_tenbis") {
+            this.setState({tenbis: event_value});
+        }
+        else if (event_id === "restaurant_maxDeliveryTime") {
+            this.setState({max_delivery_time: event_value});
+        }
+        else if (event_id == "cuisine_id") {
+            this.setState({cuisine: event_value});
+        }
+    }
+
+    newRestaurantSubmit(e) {
+        console.log("rest cuisine: " + this.state.cuisine);
+
+        fetch('/restaurants', {
+            method: 'post',
+            body: JSON.stringify({restaurant: {
+                    name: this.state.name,
+                    cuisine: this.state.cuisine,
+                    address: this.state.address,
+                    max_delivery_time: this.state.max_delivery_time,
+                    tenbis: this.state.tenbis
+                },}),
+            headers: {
+                'Content-Type': 'application/json',
+                'X-CSRF-Token': Rails.csrfToken()
+            },
+            credentials: 'same-origin'
+        }).then(function(response) {
+            console.log("success!!!");
```

Comment:
I think you can remove this line. You can also chain a "catch" to the promise chain, and block the form from closing if an exception is thrown: `.catch(error => console.error('Error:', error))`